sources/packages/web/src/views/aest/institution/InstitutionDetails.vue

```diff
@@ -0,0 +1,133 @@
+<template>
+  <h5 class="text-muted">
+    <a @click="goBack()">
+      <v-icon left> mdi-arrow-left </v-icon> Back to Institution</a
+    >
+  </h5>
+  <full-page-container>
+    <h2 color="primary-color" class="mb-15">
+      {{ initialValue.operatingName }}
+      <designation-status-badge designationStatus="&#10004; DESIGNATED" />
+    </h2>
+    <!-- TODO: Replace v-btn with vuetify2 equivalent v-tab with icon once veutify3 is released-->
+    <v-btn text variant="outlined" @click="showProfile(institutionId)"
+      ><v-icon size="25" class="mr-2">mdi-city</v-icon>Profile</v-btn
+    >
+    <v-btn text variant="outlined" @click="showPrograms(institutionId)">
+      <v-icon size="25" class="mr-2">mdi-book-open-outline</v-icon
+      >Programs</v-btn
+    >
+    <v-btn text variant="outlined" @click="showLocations(institutionId)"
+      ><v-icon size="25" class="mr-2">mdi-map-marker-outline</v-icon
+      >Locations</v-btn
+    >
+    <v-btn text variant="outlined" @click="showUsers(institutionId)"
+      ><v-icon size="25" class="mr-2">mdi-account-group-outline</v-icon
+      >Users</v-btn
+    >
+    <v-btn text variant="outlined" @click="showDesignation(institutionId)"
+      ><v-icon size="25" class="mr-2">mdi-certificate-outline</v-icon
+      >Designation</v-btn
+    >
+    <v-btn text variant="outlined" @click="showRestrictions(institutionId)"
+      ><v-icon size="25" class="mr-2">mdi-close-circle-outline</v-icon
+      >Restrictions</v-btn
+    >
+    <v-btn text variant="outlined" @click="showNotes(institutionId)"
+      ><v-icon size="25" class="mr-2">mdi-clipboard-outline</v-icon>Notes</v-btn
+    >
+    <hr />
+    <router-view v-slot="{ Component }">
+      <keep-alive>
+        <component :is="Component" />
+      </keep-alive>
+    </router-view>
+  </full-page-container>
+</template>
+
+<script lang="ts">
+import { onMounted, ref } from "vue";
+import { useRouter } from "vue-router";
+import { InstitutionService } from "@/services/InstitutionService";
+import { AESTRoutesConst } from "@/constants/routes/RouteConstants";
+import FullPageContainer from "@/components/layouts/FullPageContainer.vue";
+import DesignationStatusBadge from "@/components/generic/DesignationStatusBadge.vue";
+import { BasicInstitutionInfo } from "@/types";
+export default {
+  components: { FullPageContainer, DesignationStatusBadge },
+  props: {
+    institutionId: {
+      type: Number,
+      required: true,
+    },
+  },
+  setup(props: any) {
+    const router = useRouter();
+    const initialValue = ref({} as BasicInstitutionInfo);
+    const goBack = () => {
+      router.push({
+        name: AESTRoutesConst.SEARCH_INSTITUTIONS,
+      });
+    };
+    const showProfile = (institutionId: number) => {
```

Comment:
All these methods to the route are going to look the same. It could be only one method that receives the _name_ as a parameter.
The _institutionId_ does not be provided as a parameter since it is also available on props.

Reply:
yes you are right